types/src/schemas/stoatConfigSchema.json

```diff
@@ -129,6 +132,30 @@
         }
       }
     },
+    "metric_plugin_map": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/metric_plugin"
+      }
+    },
+    "metric_plugin": {
+      "type": "object",
+      "required": ["filename"],
+      "properties": {
+        "metadata": {
+          "type": "object",
+          "additionalProperties": true
+        },
+        "unit": {
+          "type": "string",
+          "description": "The unit will be shown in the y-axis title in the metric chart."
+        },
+        "filename": {
+          "type": "string",
+          "description": "The input file including the metric value. The file should be in JSON with these keys: \"value\" (required, number) and \"group\" (optional, string). The \"group\" key is used to group the metric values in the chart. This file will be consumed by the Stoat action as inputs for the metric task."
```

Comment:
Maybe `name` instead of `group`?

Reply:
Agree that `name` is more straightforward.

One issue is that currently all plugins have the `name` under the `metadata` field. Maybe we should move it out of `metadata` and make it a first level field for all plugins.

Reply:
Well actually why does this specify a name at all? Shouldn't it default to the task id and then fall back to the `metadata.name` instead to match the others?

Reply:
One chart can have multiple groups. So a metric task can have multiple metric group names. The example is that the job runtime has multiple jobs.

Reply:
Shouldn't the the chart rendering choose which metrics to display? It's a display configuration, not something related to the metric itself. The metric by itself as reported shouldn't have to know that it's going to be displayed on one or more chart?

Presumably the thing that renders the chart will need to reference which metrics it's tracking by task id, and when the chart renders the task id -> `metadata.name` fallback seems to make sense?

If we want to be able to group things, would it make more sense to allow multiple "tags" (like DataDog) and allow charting by all metrics it's received for a specific tag?

Reply:
Good point about tags and separating data intake and chart display.

What about this?

```yaml
metric:
  # default input json file
  filename: data_input.json
  # separate the data intake and chart layout;
  # the tags field is about the datasets
  tags:
    metric1:
      unit: second
    metric2:
      unit: day
  # the charts field is about how to group and draw the data
  charts:
    chart1:
      name: Chart 1
      tags: |
        metric1
        metric2
    chart2:
      datasets: metric1
```

Sample data input json:

```json
[
  { value: 1, tags: ["metric1", "metric2"] },
  { value: 2, tags: ["metric1"] },
  { value: 3, tags: ["metric2"] },
]
```